submissions/natashafir/friends-app/index.js

```diff
@@ -0,0 +1,90 @@
+const URL = 'https://randomuser.me/api/?results=50';
+const CARDS = document.querySelector('.cards');
+let filterBy = 'all';
+let allFriends = [];
+let friends = [];
+let sortedFriends;
+const input = document.querySelector('input');
+
+function initialApp() {
+     fetch(URL)
+        .then(response => response.json())
+        .then((data) => (allFriends = data.results))
+        .then(() => userTemplate(allFriends))
+        .then(()=> renderUsers(allFriends))
+        .catch(err => {
+            console.log(err);
+            alert('ERROR in fetching the url');
+        })
+}
+
+function userTemplate(array) {
+    allFriends = array.map(user => {
+            return {
+                name: `${user.name.first} ${user.name.last}`,
+                age: user.dob.age,
+                gender: user.gender,
+                photo: user.picture.large
+            }
+        });
+}
+
+function createCardItem(item) {
+    const template = `<div class="card-wrapper"><img class="img-card" src="${item.photo}"></img><p>${item.name}</p><p>Age: ${item.age}</p></div>`
+    return template;
+};
+
+document.querySelector(".gender").addEventListener("click", filterByChoose);
+document.querySelector(".age").addEventListener("click", filterByChoose);
+document.querySelector(".name").addEventListener("click", filterByChoose);
+input.addEventListener('input', filterByChoose);
+
+function filterByChoose({target}) {
+    let sortedArr = allFriends;
+    if (target.type != 'radio') {
+        sortedArr = sortedArr.filter(element =>
+            element.name.toLowerCase().includes(target.value.toLowerCase()));
+    }
+    sortedArr = getSortedFriends(sortedArr, target.value);
```

Comment:
The function called as `filter`, but what you also doing sorting here.

So just filter :)